doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,328 @@
+
+## 2.4. Physical system monitoring and control
+
+Distributed networks of sensors (torque, rotary position...) and actuators
+(motors, solenoids...) can be employed in certain control applications, such as
+automotive systems, robotics, or Industry 4.0. In particular, the nodes need to:
+
+  - Collect and send data with a low latency, or at least a well synchronized
+    timestamp.
+  - Potentially run control algorithms themselves.
+  - Have the timing precision to support time sensitive control.
+
+## 2.5. Edge systems for building management and automation
+
+Various sensing (light, temperature, humidity...) and environmental control
+tasks (heating, ventilation, access control...) can be done by edge nodes in
+buildings. In particular, the nodes need to be able to:
+
+  - Connect to building management system components from a range of vendors
+    via a range of domain specific protocols, such as BACnet and Modbus.
+  - Integrate with in-house cyber security management systems.
+
+## 2.6. Smart home
+
+Smart home use cases have monitoring and control aims which overlap with
+commercial building management. However, in the smart home there is less
+technical equipment, different protocols, and different system management
+criteria. In particular, the nodes here need to be able to:
+
+  - Connect to home networking equipment from a range of vendors via common
+    protocols for constrained devices in the home, such as BLE, 802.15.4 and
+    WiFi.
+  - Ensure the privacy of end users in an easy-to-use fashion.
+
+## 2.7. Daughterboards
+
+Plug-in boards can give devices immediate support for a protocol or standard,
+or let them outsource a task from the main processor. In particular, this
+requires the board to:
+
+  - Support on-chip low-level wired communication.
+
+## 2.8. Education
+
+The broad technical scope of RIOT makes it useful as a basis for education.
+In particular, this requires:
+
+  - The presence of didactic materials related to RIOT.
+  - The presence of tooling suitable to a classroom context.
+
+# 3. Design philosophies
+
+Below are the design philosophies that are typically followed by developers to
+cater for the above use cases. The sections below include descriptions of
+tradeoffs between the philosophies, and where the resolutions typically fall.
+
+## Suitability for constrained devices
+
+"Constrained" means that available memory, energy, and processor cycles are so
+reduced as to become a dominant consideration in design requirements [1]. 
+
+#### Energy efficiency
+
+RIOT nodes sometimes need to last for several years without external power, so
+they need to manage energy carefully. RIOT's tickless scheduler lets devices
+sleep while they aren't active.
+
+Modules outside the core should leverage the benefits and address the
+programming challenges of such a scheduler, without demanding that users do the
+same. An idling device should conserve energy wherever possible, by default.
```

Comment:
this sentence is difficult to parse.
How about splitting it in 2 parts? Maybe something like
1. developers of modules should...
2. users should be able to benefit from RIOT's energy efficiency mechanisms without having to cope with scheduler details or setting power modes explicitly.

Reply:
So eg:  
Developers of modules outside the core should leverage the benefits and address the
programming challenges of such a scheduler. An idling device should conserve energy wherever possible, by default. For case specific power management, appropriate control should be available to the user. This should not include having to cope with scheduler details or setting power modes explicitly.

Reply:
changed to the above, let me know if not good